### PR TITLE
feat: add C `ndarray` API and refactor `blas/ext/base/dssumpw`

### PR DESCRIPTION
**Pull request by headlessNode**
## Description

> What is the purpose of this pull request?

This pull request:

-   add C `ndarray` API and refactor `blas/ext/base/dssumpw`

## Related Issues

> Does this pull request have any related issues?

No. 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by headlessNode** [comment]
CI failure from a downstream test suite.

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/README.md [hunk 1]
```diff
@@ -81,25 +80,24 @@ var v = dssumpw( 4, x1, 2 );
...
-var N = x.length;
 
-var v = dssumpw.ndarray( N, x, 1, 0 );
+var v = dssumpw.ndarray( x.length, x, 1, 0 );
 // returns 1.0
 ```
 
 The function has the following additional parameters:
 
 -   **offset**: starting index for `x`.
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/README.md]:
```suggestion
-   **offsetX**: starting index for `x`.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/README.md [hunk 2]
```diff
@@ -147,6 +146,123 @@ console.log( v );
...
+
+#### stdlib_strided_dssumpw( N, \*X, strideX )
+
+Computes the sum of single-precision floating-point strided array elements using pairwise summation with extended accumulation and returning an extended precision result.
+
+```c
+const float x[] = { 1.0f, 2.0f, 3.0f, 4.0f };
+
+double v = stdlib_strided_dssumpw( 4, x, 1 );
+// returns 1.0
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/README.md]: The return value is incorrect here and below.

Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/README.md]:
```suggestion
// returns 10.0
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumpw/README.md [hunk 3]
```diff
@@ -147,6 +146,123 @@ console.log( v );
...
+
+#### stdlib_strided_dssumpw_ndarray( N, \*X, strideX, offsetX )
+
+Computes the sum of single-precision floating-point strided array elements using pairwise summation with extended accumulation and alternative indexing semantics and returning an extended precision result.
+
+```c
+const float x[] = { 1.0f, 2.0f, 3.0f, 4.0f };
+
+double v = stdlib_strided_dssumpw_ndarray( 4, x, 1, 0 );
+// returns 1.0
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dssumpw/README.md]:
```suggestion
// returns 10.0
```

**Review by kgryte — APPROVED** [review]
Other than the return values, everything LGTM. Thanks, @headlessNode!

**Pull request merged**